snuba/query/parser/expressions.py

```diff
@@ -1,28 +1,78 @@
+import re
+
 from typing import Any, Optional
 
-from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.util import is_function, QUOTED_LITERAL_RE
 
+# We still have some cases were people pass clickhouse expressions in the
+# aggregate clause of the Snuba query since snuba does not reject them.
+# In order to process the aggregate first we need to turn those expressions
+# into Snuba functions.
+# This regex provides some very basic support to decompose those expressions.
+AGGR_CLICKHOUSE_FUNCTION_RE = re.compile(
+    r"^(?P<name>[a-zA-Z_][a-zA-Z0-9_]*)\s*\((?P<params>[\w\,\.\s]*)\)$"
```

Comment:
See the test attached to see how this works.

Reply:
Why do we allow whitespace?

Reply:
right, it should not be allowed. Will fix